src/Simulation/Simulators.Tests/OpenSystemsSimulatorTests/TestOperations.qs

```diff
@@ -0,0 +1,197 @@
+// to help build confidence in shared decompositions, further improving test
+// coverage.
+
+namespace Microsoft.Quantum.Experimental.Tests {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Diagnostics;
+
+    internal function Fact(expected : Bool, message : String) : Unit {
+        if not expected {
+            fail $"Fact was false: {message}";
+        }
+    }
+
+    internal operation MX(target : Qubit) : Result {
+        return Measure([PauliX], [target]);
+    }
+
+    @Test("Microsoft.Quantum.Experimental.OpenSystemsSimulator")
+    operation CheckBellBasisParitiesWithSingleQubitMeasurements() : Unit {
+        use (left, right) = (Qubit(), Qubit()) {
+            H(left);
+            CNOT(left, right);
+
+            Fact(M(left) == M(right), "Z parity in 00 case was wrong.");
+            ResetAll([left, right]);
+        }
+
+        use (left, right) = (Qubit(), Qubit()) {
+            H(left);
+            CNOT(left, right);
+
+            Fact(MX(left) == MX(right), "X parity in 00 case was wrong.");
+            ResetAll([left, right]);
+        }
+
+        use (left, right) = (Qubit(), Qubit()) {
+            H(left);
+            CNOT(left, right);
+
+            X(left);
+
+            Fact(M(left) != M(right), "Z parity in 10 case was wrong.");
+            ResetAll([left, right]);
+        }
+
+        use (left, right) = (Qubit(), Qubit()) {
+            H(left);
+            CNOT(left, right);
+
+            X(left);
+
+            Fact(MX(left) == MX(right), "X parity in 10 case was wrong.");
+            ResetAll([left, right]);
+        }
+
+        use (left, right) = (Qubit(), Qubit()) {
+            H(left);
+            CNOT(left, right);
+
+            Z(left);
+
+            Fact(M(left) == M(right), "Z parity in 01 case was wrong.");
+            ResetAll([left, right]);
+        }
+
+        use (left, right) = (Qubit(), Qubit()) {
+            H(left);
+            CNOT(left, right);
+
+            Z(left);
+
+            Fact(MX(left) != MX(right), "X parity in 01 case was wrong.");
+            ResetAll([left, right]);
+        }
+
+        use (left, right) = (Qubit(), Qubit()) {
+            H(left);
+            CNOT(left, right);
+
+            X(left);
+            Z(left);
+
+            Fact(M(left) != M(right), "Z parity in 11 case was wrong.");
+            ResetAll([left, right]);
+        }
+
+        use (left, right) = (Qubit(), Qubit()) {
+            H(left);
+            CNOT(left, right);
+
+            X(left);
+            Z(left);
+
+            Fact(MX(left) != MX(right), "X parity in 11 case was wrong.");
+            ResetAll([left, right]);
+        }
+    }
+
+    internal function Xor(a : Bool, b : Bool) : Bool {
+        return (a or b) and ((not a) or (not b));
+    }
+
+    @Test("Microsoft.Quantum.Experimental.OpenSystemsSimulator")
+    @Test("QuantumSimulator") // validate against full-state simulator.
```

Comment:
This is the only test case that also gets run against the full-state simulator. Besides overall pipeline perf, is there any reason the other tests shouldn't be run against the full-state simulator too?

Reply:
Was indeed thinking of overall pipeline performance, given we routinely hit up against the limit for job time. That said, adding more validations against full-state simulator shouldn't cost that much time, so let me go on and add that.